core/src/com/unciv/logic/city/CityStats.kt

```diff
@@ -178,15 +178,17 @@ class CityStats {
         if (!civInfo.isPlayerCivilization())
             unhappinessModifier *= civInfo.gameInfo.getDifficulty().aiUnhappinessModifier
 
-        var unhappinessFromCity = -3f     // -3 happiness per city
+        var unhappinessFromCity = -3f // -3 happiness per city
         if (civInfo.hasUnique("Unhappiness from number of Cities doubled"))
-            unhappinessFromCity *= 2f//doubled for the Indian
+            unhappinessFromCity *= 2f //doubled for the Indian
 
         newHappinessList["Cities"] = unhappinessFromCity * unhappinessModifier
 
         var unhappinessFromCitizens = cityInfo.population.population.toFloat()
-        if (civInfo.hasUnique("Specialists produce half normal unhappiness"))
-            unhappinessFromCitizens -= cityInfo.population.getNumberOfSpecialists() * 0.5f
+        var SpecialistUnhappinessReductionModifier = civInfo.getMatchingUniques("Specialists only produce []% of normal unhappiness")
```

Comment:
Also triggers the "LocalVariableName" inspection. And we get 1% reduction for free (the fold initializer - I wouldn't be surprised there's a reason but that would merit a comment). And - I dislike implicit casts (on SQL Server it's definitely worth it to pay attention to those) - anyway, the `1` two lines down (and in lines you didn't touch) is implicitly cast to Float and I prefer having those in the open - just a personal quirk, I won't mention it again.